modules/machineset-gcp-enabling-gpu-support.adoc

```diff
@@ -0,0 +1,125 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc
+
+:_content-type: PROCEDURE
+[id="machineset-gcp-enabling-gpu-support_{context}"]
+= Enabling GPU support for a machine set
+
+Google Cloud Platform (GCP) Compute Engine allows users to add GPUs to VM instances. Workloads that benefit from access to GPU resources can perform better on compute machines with this feature enabled. {product-title} on GCP supports NVIDIA GPU models in the A2 and N1 machine series.
+
+.Supported GPU configurations
+|====
+|Model name |GPU type |Machine types ^[1]^
+
+|NVIDIA A100
+|`nvidia-tesla-a100`
+a|* `a2-highgpu-1g`
+* `a2-highgpu-2g`
+* `a2-highgpu-4g`
+* `a2-highgpu-8g`
+* `a2-megagpu-16g`
+// Of these, only `a2-highgpu-1g` is mentioned in the test plan. Should we assume all of these types work, or not?
+
+|NVIDIA K80
+|`nvidia-tesla-k80`
+.5+a|* `n1-standard-1`
+* `n1-standard-2`
+* `n1-standard-4`
+* `n1-standard-8`
+* `n1-standard-16`
+* `n1-standard-32`
+* `n1-standard-64`
+* `n1-standard-96`
+// Of these, only `n1-standard-1` with `nvidia-tesla-p100` is mentioned in the test plan. Should we assume all of these combinations work, or not?
+
+|NVIDIA P100
+|`nvidia-tesla-p100`
+
+|NVIDIA P4
+|`nvidia-tesla-p4`
+
+|NVIDIA T4
+|`nvidia-tesla-t4`
+
+|NVIDIA V100
+|`nvidia-tesla-v100`
+
+|====
+[.small]
+--
+1. For more information about machine types, including specifications, see the GCP Compute Engine documentation about link:https://cloud.google.com/compute/docs/general-purpose-machines#n1_machines[N1 machine series] and link:https://cloud.google.com/compute/docs/accelerator-optimized-machines#a2_vms[A2 machine series].
+--
+
+You can define which supported GPU to use for an instance by using the Machine API.
+
+You can configure machines in the N1 machine series to deploy with a GPU attached, or to use a guest accelerator.
+
+Machines in the A2 machine series come with associated GPUs, and cannot use guest accelerators.
+
+[NOTE]
+====
+GPUs for graphics workloads are not supported.
+====
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Specify the number and type of GPUs under the `providerSpec` field in your machine set YAML file. See the following examples of valid configurations:
++
+.Example configuration for the A2 machine series:
+[source,yaml]
+----
+  providerSpec:
+    value:
+      gpus:
+      - count: 1 <1>
+        type: nvidia-tesla-a100 <2>
```

Comment:
With A2 family machine types `nvidia-tesla-a100` GPUs are already attached therefore should not be specified by user in `providerSpec`. 
`gpus` field should not be specified in this case. Number of gpus included with each machine type is in its name. But also it would be nice to include the link. https://cloud.google.com/compute/docs/gpus#a100-gpus

MAPI code link https://github.com/openshift/machine-api-provider-gcp/blob/main/pkg/cloud/gcp/actuators/machine/reconciler.go#L119

Reply:
Ok, I think I gather that this should be more like:
```
----
  providerSpec:
    value:
      machineType: a2-highgpu-1g <3>
      onHostMaintenance: Terminate <4>
      restartPolicy: Always <5>
----
```
But if that's still not quite right, I will wait until Polarion is up and check against the test case info.